src/csharp/Grpc.HealthCheck/HealthGrpc.cs

```diff
@@ -38,11 +38,22 @@ public interface IHealthClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IHealth
     {
       Task<global::Grpc.Health.V1.HealthCheckResponse> Check(global::Grpc.Health.V1.HealthCheckRequest request, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class HealthBase
```

Comment:
If we delay this change to also fix the client-side in the same PR (which seems like it would be a breaking change and there's no way around it), we might be able to kill the serverside interface immediately (also see my other comment about adding Grpc suffix to the umbrella class).